source/admin_manual/importing_mailboxes.rst

```diff
@@ -55,3 +55,81 @@ If you have only read privileges, you can try using in-memory indexes.
 
   doveadm import -u destuser maildir:/opt/backup/destuser/Maildir:INDEX=MEMORY "" ALL
 
+
+Merging Storages
+----------------
+
+In some disaster recovery cases you may end up having mails for the same user
+in two different locations, and need to merge them. For example the storage
+goes down and fixing it takes a long time, so during the fixing you can let the
+users access their emails as an empty account, which can receive new mails.
+Later on you can use ``doveadm import`` to merge the mailboxes.
+
+Note that there is no way to make this solution perfect:
+ * IMAP clients that have cached mails locally will delete their local caches
+   and have to re-download mails later on.
+ * POP3 clients that leave mails on server will notice all the mails are gone,
+   and delete their local UIDL caches. When old mails come back, they're
+   re-downloaded as new emails (duplicates).
+
+Also, there are 3 alternative ways of how mails can be imported into mailboxes:
+ * Old recovered mails are imported on top of the newly received mails. The
+   downside here is that mails may now be sorted in a weird order. If the IMAP
+   client shows the mails in the saved order, the new received emails show up
+   as oldest emails. Although this may not be an issue, since many IMAP clients
+   sort the mails by either Date: header or the received timestamp (IMAP
+   INTERNALDATE).
+ * New mails are imported on top of the old recovered mails. This avoids the
+   sorting problems, so it's likely the preferred method. The downside here is
+   that the IMAP/POP3 clients will have to re-download also the newly delivered
+   emails, as well as the old ones. Another issue with this is that some IMAP
+   clients might not show the old recovered mails without manually rebuilding
+   local caches, because the mails become inserted to the beginning of the
+   folders, which isn't allowed by the IMAP protocol.
+
+     * Another thing to keep mind mind here is that IMAP clients shouldn't see
```

Comment:
Duplicate `mind`.

Reply:
oops, and already merged the approved 2.3 cherry-pick, need to fix there also..